README.md

```diff
@@ -733,6 +733,12 @@ The following typedefs are not generated -
   - They refer to a struct/union having the same name as itself.
   - They refer to a boolean, enum, inline array, Handle or any unsupported type.
 
+### How are macros handles?
+
+An understanding of your program's source headers is beyond `ffigen`'s perview; rather, it uses `clang`'s own compiler frontend to parse and traverse the `C` header files. As macros are an advanced feature of the compiler, `ffigen` needs to first expand the macros using `clang`'s macro expansion and then run its own traversal. To do this, `ffigen` generates temporary files in a system tmp directory.
```

Comment:
`ffigen` uses `clang`'s own compiler frontend to parse and traverse the `C` header files. `ffigen` expands the macros using `clang`'s macro expansion and then traverses the expanded code. To do this, `ffigen` generates temporary files in a system tmp directory.